src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -302,7 +390,72 @@ TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extent(
     cache->duplicate_for_write(t, extent);
     return rewrite_extent_iertr::now();
   }
-  return lba_manager->rewrite_extent(t, extent);
+
+  return lba_manager->rewrite_physical_extent(t, extent);
+}
+
+TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extents(
+  Transaction &t,
+  std::vector<CachedExtentRef>& extents)
+{
+  LOG_PREFIX(TransactionManager::rewrite_extents);
+  std::vector<LogicalCachedExtentRef> logical_extents;
+  return seastar::do_with(
+    std::vector<LogicalCachedExtentRef>(),
+    [this, &extents, &t, FNAME](auto& logical_extents) mutable {
+    return trans_intr::do_for_each(extents,
+      [this, &logical_extents, &t, FNAME](CachedExtentRef& extent) mutable
+      -> rewrite_extent_ret {
+      if (extent->is_logical()) {
+	logical_extents.emplace_back(extent->cast<LogicalCachedExtent>());
+	return seastar::now();
+      } else {
```

Comment:
If we think it is not worth dealing with ool physical extents right now, a proper way might be to hard-code the hint of physical extents to be always allocated as inline extents at allocation time. And leave the rest work in the general `submit_transaction()` path.

Reply:
I think we have to first settle whether the extent placement manager should take the journal into account as well. As I said [here](https://github.com/ceph/ceph/pull/42270#issuecomment-882980852), the whole idea of the extent placement manager is about choosing the right address for ool extents to be written to, and I don't see much choice when it comes to journal records. I think for journal records, they just go to journal segments, making them going through the extent placement manager looks not consistent with the philosophy of the extent placement manager to me and doesn't seem to offer much advantages. Am I right?

Reply:
I didn't mean to modify the behavior of the existing `SegmentedJournal`. The `ExtentWriter` for inline extents can be noop. The `SegmentedInlineAllocator` construct the concreate type of `CachedExtent`, and `Cache` can then assign a relative address if the extent is decided inlined.

Reply:
> I didn't mean to modify the behavior of the existing `SegmentedJournal`. The `ExtentWriter` for inline extents can be noop. The `SegmentedInlineAllocator` construct the concreate type of `CachedExtent`, and `Cache` can then assign a relative address if the extent is decided inlined.

Oh, sorry, I didn't make myself clear. By `"whether the extent placement manager should take the journal into account"`, I mean whether the extent placement manager should take journal records into account. And I think what you mean is the extent placement manager should decide whether an extent is inline or out-of-line. Am I right?

Reply:
> And I think what you mean is the extent placement manager should decide whether an extent is inline or out-of-line. Am I right?

Exactly, but this doesn't conform with what @athanatos was suggested, so this would need his reviews.